XenModel/Actions/VM/InstallPVToolsAction.cs

```diff
@@ -134,11 +131,8 @@ private XenAPI.VDI findWinISO(bool searchHiddenISOs)
             {
                 if (XenAPI.SR.SRTypes.iso.ToString() == sr.content_type)
                 {
-                    foreach (VDI vdi in Connection.ResolveAllShownXenModelObjects(sr.VDIs, searchHiddenISOs))
-                    {
-                        if (vdi.IsToolsIso)
-                            return vdi;
-                    }
+                    var vdis = Connection.ResolveAllShownXenModelObjects(sr.VDIs, searchHiddenISOs);
+                    return vdis.FirstOrDefault(v => v.IsToolsIso);

```

Comment:
I think this will return null for the first iso-SR (if no VDI is IsToolsIso) and will not check further SRs